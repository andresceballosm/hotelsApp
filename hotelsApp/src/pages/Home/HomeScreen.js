import React, { Component } from 'react';
import {
  View,
  TextInput,
  Text,
} from 'react-native';
import { connect } from 'react-redux';
import { ButtonGeneral } from '../../components/Button';
import { ActionSetLoading, ActionSetCity } from '../../store/actions/ActionApp';
import { ActionGetHotelByParameter } from '../../store/actions/ActionHotels';
import styles from './Styles';

class HomeScreen extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      searchText:'',
    };
};
  searchText = (e) => {
      this.setState({searchText : e})
  }
  render(){
    const { navigation } = this.props;
    return (
        <View style={styles.container}>
          <View style={styles.viewTitle}>
            <Text style={styles.title}>Â¿A donde quieres viajar?</Text>
          </View>
          <TextInput
          paddingLeft={12}
          style={styles.searchBarHome}
          value={this.state.searchText}
          onChangeText={value => this.searchText(value)}
          placeholder={'Buscar por ciudad EJ: Bogota, Medellin'} />
          <View style={styles.btn}>
            <ButtonGeneral
            title="Buscar" 
            click={() => { 
                this.props.searchHotelsByCity(this.state.searchText, navigation)
            }}
            color="#db6e2e"
            fontColor="white"/>
          </View>
        </View>
    );
  }
};

const mapStateToProps = state => ({
});

const mapDispatchToProps = dispatch => ({
  searchHotelsByCity:(city, navigation) => {
    dispatch(ActionSetLoading());
    dispatch(ActionSetCity(city));
    let parameter = 'city';
    let value = city;
    dispatch(ActionGetHotelByParameter(parameter, value));
    navigation.navigate('hotels')
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
