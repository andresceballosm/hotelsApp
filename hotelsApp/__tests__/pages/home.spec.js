import React from 'react';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import HomeScreen from '../../src/pages/Home/HomeScreen';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });
const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  preferences: {
    hotels: false,
  },
};

describe('Testing HomeScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
    <Provider store={mockStore(initialState)}>
      <HomeScreen /> 
    </Provider>,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});