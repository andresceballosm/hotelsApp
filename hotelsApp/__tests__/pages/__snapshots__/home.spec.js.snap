// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen renders as expected 1`] = `ShallowWrapper {}`;
